splink/linker.py

```diff
@@ -262,6 +262,29 @@ def _get_input_columns(
 
         return column_names
 
+    @property
+    def _column_names_as_input_columns(
```

Comment:
This appears to be getting duplicated all over the place.

Ideally we'll have a single method that handles the extraction of our input columns - https://github.com/moj-analytical-services/splink/pull/1611, that can then be extended to clean, process, etc.

Reply:
I'm happy to approve this PR with this function in it, if you're ok with cleaning it up once Andy's original PR goes in?

Reply:
I've merged in `master` that includes Andy's work, and added arguments to his function so it's easy to filter out unique id columns and any specified `_additional_columns_to_retain`